docs/src/threadpool.rst

```diff
@@ -32,35 +55,156 @@ Data types
 
 .. c:type:: void (*uv_work_cb)(uv_work_t* req)
 
-    Callback passed to :c:func:`uv_queue_work` which will be run on the thread
-    pool.
+    Callback passed to :c:func:`uv_queue_work` which will be run on
+    the executor.
 
 .. c:type:: void (*uv_after_work_cb)(uv_work_t* req, int status)
 
     Callback passed to :c:func:`uv_queue_work` which will be called on the loop
-    thread after the work on the threadpool has been completed. If the work
-    was cancelled using :c:func:`uv_cancel` `status` will be ``UV_ECANCELED``.
+    thread after the work on the executor has been completed or cancelled.
+    If the work was cancelled using :c:func:`uv_cancel`,
+    then `status` will be ``UV_ECANCELED``.
+
+.. c:type:: uv_executor_t
+
+    Executor type. Use when overriding the default threadpool.
+
+.. c:type:: void (*uv_executor_init_func)(uv_executor_t* executor)
+
+    Called once, the first time :c:func:`uv_executor_queue_work` is called.
+    Do any initialization you want.
+
+.. c:type:: void (*uv_executor_destroy_func)(uv_executor_t* executor)
+
+    Called at some point ???
+
+.. c:type:: void (*uv_executor_submit_func)(uv_executor_t* executor, uv_work_t* req, uv_work_options_t* opts)
 
+    Called when the executor should handle this request.
+
+.. c:type:: int (*uv_executor_cancel_func)(uv_executor_t* executor, uv_work_t* req)
+
+    Called when someone wants to cancel a previously-submit'ed request.
+    Return ``UV_EBUSY`` if you cannot cancel it.
```

Comment:
I think mixing `someone` and `you` is a bit odd?

Reply:
Regarding `uv_executor_destroy_func`: I think the responsibility for initialization and destruction if best left to the embedder. I am planning to remove these unless someone thinks this is a good idea.

Reply:
Done.